pkg/scheduler/util/utils.go

```diff
@@ -120,6 +121,9 @@ func GetPodAntiAffinityTerms(affinity *v1.Affinity) (terms []v1.PodAffinityTerm)
 // PatchPod calculates the delta bytes change from <old> to <new>,
 // and then submit a request to API server to patch the pod changes.
 func PatchPod(cs kubernetes.Interface, old *v1.Pod, new *v1.Pod) error {
+	if reflect.DeepEqual(old, new) {
```

Comment:
Is there any place calling this function when old and new are equal?

BTW: This function seems only change pod status, so `reflect.DeepEqual(old.Status, new.Status)` might be enough.

Reply:
A couple of things:
1) DeepEqual is expensive
2) We don't call this function unless there is something to update (either updating the pod condition or clearing a set NominatedNodeName).

Reply:
one thing we can do is check `if len(patchBytes) == 0 {return nil}` below

Reply:
I'd hope quality of the public method like this doesn't depends on how it's got called even though it might not happen today. 
Have done a check that `len(patchBytes)` won't be zero even old and new are identical, maybe something is padding there, I am not sure.

Reply:
Could you check what's the output of `CreateTwoWayMergePatch` for the same object? It might be something like `{}`. There's no point on adding an extra traversal over the Pod objects.

Reply:
Here it is when they are the same object, 
`(dlv) p patchBytes
[]uint8 len: 2, cap: 8, [123,125]
`
not sure what's those bytes stand for, will check it later.

Reply:
seems like the ASCII is "S",  "U".

Reply:
It is indeed `{}`

Reply:
hmm... sorry, I cannot get your point here, why it's `{}` when it's hold two bytes there? for this case, `len(patchBytes) == 2` instead of `0`, am I wrong?

Reply:
Get what your meant,  PR is updated.

Reply:
why unmarshal instead of doing a simple string comparison?